packages/core/src/lib/markdown-it/highlight/HighlightRuleComponent.js

```diff
@@ -0,0 +1,57 @@
+const LINESLICE_REGEX = new RegExp('(\\d+)\\[(\\d*):(\\d*)]');
+
+class HighlightRuleComponent {
+  constructor(components) {
+    /**
+     * @type Number | Array<Number>
+     */
+    this.components = components;
+  }
+  
+  isNumber() {
+    return Number.isInteger(this.components);
+  }
+  
+  isSlice() {
+    return Array.isArray(this.components)
+      && this.components.length === 3
+      && this.components.every(Number.isInteger);
+  }
+  
+  isUnboundedSlice() {
+    return this.isSlice()
+      && this.components[1] === -1
```

Comment:
let's remove the use of magic numbers in favour of named instance properties, so we don't need to refer back to the usage to understand which is which